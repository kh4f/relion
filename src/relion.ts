import { bump, commit, tag, changelog } from '@/lifecycles'
import { resolveConfig, setSilent } from '@/utils'
import type { UserConfig } from '@/types'

export default async function relion(userConfig: UserConfig) {
	setSilent(!!userConfig.silent || (!!userConfig.profile && !!userConfig[`_${userConfig.profile}`]?.silent))

	const resolvedConfig = await resolveConfig(userConfig)

	const bumpResults = bump(resolvedConfig)
	const generatedChangelog = changelog(resolvedConfig)
	const commitCommand = commit(resolvedConfig)
	const tagCommand = tag(resolvedConfig)

	return { resolvedConfig, generatedChangelog, commitCommand, tagCommand, bumpResults }
}

export const defineConfig = (config: UserConfig): UserConfig => config