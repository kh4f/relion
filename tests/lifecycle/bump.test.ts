import { describe, expect, it } from 'vitest'
import { readFileSync, writeFileSync } from 'node:fs'
import relion from '@/.'

describe('version bumping', () => {
	it('should bump version in manifest.json', () => {
		const origManifestContent = readFileSync('tests/fixtures/manifest.json', 'utf8')
		relion({
			dryRun: false,
			context: { newVersion: '1.2.3' },
			bump: [{
				file: 'tests/fixtures/manifest.json',
				pattern: /(version": )".*"/,
				replacement: `$1"{{newVersion}}"`,
			}],
		})
		expect(readFileSync('tests/fixtures/manifest.json', 'utf8')).toMatchSnapshot()
		writeFileSync('tests/fixtures/manifest.json', origManifestContent, 'utf8')
	})

	it(`should append new version entry to 'versions.json'`, () => {
		const origVersionsContent = readFileSync('tests/fixtures/versions.json', 'utf8')
		relion({
			dryRun: false,
			context: { newVersion: '1.2.3' },
			bump: [{
				file: 'tests/fixtures/versions.json',
				pattern: /(.*")/s,
				replacement: `$1,\n\t"{{newVersion}}": "${/(^.*?minAppVersion": ")(.*?)(")/s.exec(readFileSync('tests/fixtures/manifest.json', 'utf8'))?.[2]}"`,
			}],
		})
		expect(readFileSync('tests/fixtures/versions.json', 'utf8')).toMatchSnapshot()
		writeFileSync('tests/fixtures/versions.json', origVersionsContent, 'utf8')
	})
})