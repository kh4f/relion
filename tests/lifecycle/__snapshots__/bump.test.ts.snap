// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`version bumping > should append new version entry to 'versions.json' 1`] = `
"{
	"1.10.1": "0.15.0",
	"1.10.2": "0.15.0",
	"1.11.0": "0.15.0",
	"1.11.1": "0.15.0",
	"1.11.2": "0.15.0",
	"1.2.3": "0.15.0"
}"
`;

exports[`version bumping > should bump version in manifest.json 1`] = `
"{
	"id": "obsidian-plugin",
	"name": "Obsidian Plugin",
	"version": "1.2.3",
	"minAppVersion": "0.15.0",
	"description": "An Obsidian plugin.",
	"author": "kh4f",
	"authorUrl": "https://github.com/kh4f",
	"isDesktopOnly": false
}"
`;
